File: app/src/main/java/com/example/votesapp/activities/lista_salas/Lista_Salas.java
package com.example.votesapp.activities.lista_salas;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListAdapter;
import android.widget.ListView;

import com.example.votesapp.R;

import java.util.ArrayList;
import java.util.List;

public class Lista_Salas extends AppCompatActivity {

    //Atributos
    private ListView slistView;
    List<Sala> sLista = new ArrayList<>();

    //Llamamos el adapter
    ListAdapter sAdapter;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lista_salas);

        //Obtener instancia de la lista
        slistView = findViewById(R.id.listView);


        //inicializamos el adaptador que va a poner la lista de sala
        sAdapter = new SalaAdapter(this);

        //Crear adaptador y setear
        slistView.setAdapter(sAdapter);


    }


}

File: app/src/main/java/com/example/votesapp/activities/lista_salas/Lista_Salas.kt
package com.example.votesapp.activities.lista_salas

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.votesapp.R
import com.example.votesapp.activities.mis_salas.MisSalas
import com.example.votesapp.activities.lista_salas.SalaAdapter
import java.util.*

class Lista_Salas : Fragment() {
    //Atributos
    private var slistView: ListView? = null

    //Llamamos el adapter
    private var sAdapter: ListAdapter? = null

    private var username : String? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val viewFragment = inflater.inflate(R.layout.listado_salas, container, false)
        slistView = viewFragment.findViewById(R.id.listView)

        username = this.activity?.intent?.getStringExtra("param_username")

        //inicializamos el adaptador que va a poner la lista de sala
        sAdapter = SalaAdapter(activity,"userVotante/$username")

        //Crear adaptador y setear
        slistView?.adapter = sAdapter

        return viewFragment
    }

    companion object {
        fun newInstance(): Lista_Salas = Lista_Salas()
    }
}